### PR TITLE
Papercut type fixes

### PR DESCRIPTION
**Pull request by geograham**
## Linked Issues

Papercut #3022

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by je-cook — CHANGES_REQUESTED** [review]
Thanks for being the guinea pig on this, I will add any stylistic decisions to the issue that come out of this PR once we've all agreed

File: bluemira/equilibria/fem_fixed_boundary/fem_magnetostatic_2D.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def psi_norm_2d(self) -> Callable[[np.ndarray], np.ndarray]:
             np.abs((self.psi(x) - self.psi_ax) / (self.psi_b - self.psi_ax))
         )
 
-    def set_mesh(self, mesh: Union[dolfin.Mesh, str]):
+    def set_mesh(self, mesh: Union[dolfin.Mesh, str]) -> None:
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/equilibria/fem_fixed_boundary/fem_magnetostatic_2D.py:164]: Ok I've had a chat, a search and a think and I think we should remove the None return type when there is no return (implicit return) like above.

File: bluemira/equilibria/boundary.py [hunk 2]
```diff
@@ -44,7 +45,7 @@ def __init__(self, grid: Grid):
             values[i] = g
         self.f_greens = values
 
-    def __call__(self, psi: np.ndarray, jtor: np.ndarray):
+    def __call__(self, psi: npt.NDArray[np.float64], jtor: npt.NDArray[np.float64]):
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/equilibria/boundary.py:48]: Urr thanks for looking into it. I feel thats a numpy problem... How about `npt.NDArray[np.float_]` as you put somewhere?

[the remark this answers is not in the document]
Comment (geograham) [reply bluemira/equilibria/boundary.py:48]: Our pre-commit seems to automatically change npt.NDArray[np.float_] to npt.NDArray[np.float64].

File: bluemira/equilibria/optimisation/constraint_funcs.py [hunk 3]
```diff
@@ -14,10 +14,10 @@
 
     class Constraint(ConstraintFunction):
 
-        def f_constraint(self, vector: npt.NDArray) -> npt.NDArray:
+        def f_constraint(self, vector: npt.NDArray[np.float_]) -> npt.NDArray[np.float_]:
```
Comment (je-cook) [inline bluemira/equilibria/optimisation/constraint_funcs.py:17]:
```suggestion
        def f_constraint(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
```

File: bluemira/equilibria/optimisation/constraint_funcs.py [hunk 4]
```diff
@@ -14,10 +14,10 @@
...
             return constraint_calc(vector)
 
-        def df_constraint(self, vector: npt.NDArray) -> npt.NDArray:
+        def df_constraint(self, vector: npt.NDArray[np.float_]) -> npt.NDArray[np.float_]:
```
Comment (je-cook) [inline bluemira/equilibria/optimisation/constraint_funcs.py:20]:
```suggestion
        def df_constraint(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
```

File: bluemira/equilibria/optimisation/objectives.py [hunk 5]
```diff
@@ -13,10 +13,10 @@
 
     class Objective(ObjectiveFunction):
 
-        def f_objective(self, vector: npt.NDArray) -> npt.NDArray:
+        def f_objective(self, vector: npt.NDArray[np.float_]) -> npt.NDArray[np.float_]:
```
Comment (je-cook) [inline bluemira/equilibria/optimisation/objectives.py:16]:
```suggestion
        def f_objective(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
```

File: bluemira/equilibria/optimisation/objectives.py [hunk 6]
```diff
@@ -13,10 +13,10 @@
...
             return objective_calc(vector)
 
-        def df_objective(self, vector: npt.NDArray) -> npt.NDArray:
+        def df_objective(self, vector: npt.NDArray[np.float_]) -> npt.NDArray[np.float_]:
```
Comment (je-cook) [inline bluemira/equilibria/optimisation/objectives.py:19]:
```suggestion
        def df_objective(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
```

**Review by je-cook — APPROVED** [review]

File: bluemira/equilibria/boundary.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (geograham) [reply bluemira/equilibria/boundary.py:48]: No worries!

**Pull request merged**